src/Message/AbstractRequest.php

```diff
@@ -317,8 +325,10 @@ public function sendData($data)
             }
         }
 
+        $eventDispatcher = $this->httpClient->getEventDispatcher();
+
         // don't throw exceptions for errors
-        $this->httpClient->getEventDispatcher()->addListener(
+        $eventDispatcher->addListener(
```

Comment:
in omnipay-vindicia, you acknowledge that $eventDispatcher could be null. any reason you're not doing that here?

Reply:
Basically it's because of Guzzle. The Guzzle client that is used by omnipay extends `AbstractHasDispatcher` so it'll always have an event dispatcher. I can definitely add checks for future proofing in case the http client is changed.

Reply:
why'd vindicia need the null checks then?

Reply:
You have a good point. The http client that Vindicia takes has ClientInterface which  extends HasDispatcherInterface so I guess it wasn't necessary.

Reply:
aight

Reply:
So ok if I leave this as is?

Reply:
yeah